### ISSUE TITLE
New rule: no-implicit-access

### ISSUE THREAD

**Issue by TrySound** [atfzl/eslint-plugin-css-modules#9]
This rule can process two cases:
1. passing or assigning imported hash with classes anywhere
2. accessing class dynamically like `styles[var_with_any_name]`, but string is valid `styles['className']`

**Comment by atfzl** [comment]
Can you provide example for both cases.

**Comment by TrySound** [comment]
```js
import css from './module.css';

const processStyles = styles => {
   return styles;
};

// passes whole hash directly -> can't be linted
console.log(processStyles(css));

// Assigns whole hash directly -> can't be linted
const styles = css;

console.log(styles);
```

```js
import css from './module.css';

let className = 'table';

// Result is not predictable
console.log(css[className]);

// valid
console.log(css['table']);
```

**Comment by atfzl** [comment]
Ok, it's more clear now.

Let's summarise what is needed:

## Case 1:
We will show lint error if the default import variable (like `css` in `import css from './module.css'`):
* is passed to a function as parameter
```js
const processStyles = styles => {
   return styles;
};

console.log(processStyles(css)); // lint error here
```
or
* is assigned to some other variable
```js
const styles = css; // lint error here
```

## Case 2:
Give error when accessing dynamic computed properties.

Example:
```js
import css from './module.css';

let className = 'table';

console.log(css[className]); // lint error here

console.log(css['table']); // lint ok
```


I think both these cases should be different rules.
